projects/components/src/datagrid/datagrid.component.html

```diff
@@ -88,9 +88,15 @@ <h3 *ngIf="this.header" class="vcd-header">{{ header }}</h3>
             >
             </ng-template>
         </clr-dg-cell>
-        <ng-container ngProjectAs="clr-dg-row-detail" *ngIf="detailTemplate !== undefined">
+        <ng-container ngProjectAs="clr-dg-row-detail" *ngIf="detailComponent !== undefined">
             <clr-dg-row-detail *clrIfExpanded>
-                <ng-content *ngTemplateOutlet="detailTemplate; context: { record: restItem }"> </ng-content>
+                <ng-template
+                    [vcdComponentRendererOutlet]="{
+                        rendererSpec: getDetailRenderSpec(restItem, i, count),
+                        provider: getRowLoadingListenerInjector(i)
```

Comment:
Let's avoid the double loop here by getting the listener injector with a custom component instead. This custom component should not be exported from its module.

Reply:
Yes. I think there should be a separate component:
- Perhaps called RowDetailComponent
- This component takes a XyzDetailComponent constructor(given as input to DataGrid) and it's context(rest item)
- This component then gets/creates the desired provider from the corresponding row around it
- And this component then passes the XyzDetailComponent constructor, it's context(rest item) and the provider extracted to componentRendererOutlet directive as:
```
{
 rendererSpec: {type: Type<XyzDetailComponent>},
 context: restItem,
 injector?: <extractedInjector>
}
```

Reply:
I tried to take this approach as best as I could, let me know what you think!